larpandoracontent/LArThreeDReco/LArTwoViewMatching/TwoViewLongTracksTool.cc

```diff
@@ -0,0 +1,173 @@
+    const unsigned int minMatchedSamplingPointRatio, const pandora::ClusterSet &usedClusters)
+{
+    const unsigned int nMatchedReUpsampledSamplingPoints((*iIter)->GetOverlapResult().GetNMatchedReUpsampledSamplingPoints());
+
+    for (MatrixType::ElementList::const_iterator eIter = elementList.begin(); eIter != elementList.end(); ++eIter)
+    {
+        if ((*iIter) == eIter)
+            continue;
+
+        if (usedClusters.count(eIter->GetCluster1()) || usedClusters.count(eIter->GetCluster2()))
+            continue;
+
+        if (((*iIter)->GetCluster1() != eIter->GetCluster1()) && ((*iIter)->GetCluster2() != eIter->GetCluster2()))
+            continue;
+
+        if (nMatchedReUpsampledSamplingPoints < minMatchedSamplingPointRatio * eIter->GetOverlapResult().GetNMatchedReUpsampledSamplingPoints())
+            return false;
+    }
+
+    return true;
+}
+
+//------------------------------------------------------------------------------------------------------------------------------------------
+
+bool TwoViewLongTracksTool::Run(TwoViewTransverseTracksAlgorithm *const pAlgorithm, MatrixType &overlapMatrix)
+{
+   if (PandoraContentApi::GetSettings(*pAlgorithm)->ShouldDisplayAlgorithmInfo())
+       std::cout << "----> Running Algorithm Tool: " << this->GetInstanceName() << ", " << this->GetType() << std::endl;
+
+    ProtoParticleVector protoParticleVector;
+    this->FindLongTracks(overlapMatrix, protoParticleVector);
+
+    const bool particlesMade(pAlgorithm->CreateThreeDParticles(protoParticleVector));
+    return particlesMade;
+}
+
+//------------------------------------------------------------------------------------------------------------------------------------------
+
+void TwoViewLongTracksTool::FindLongTracks(const MatrixType &overlapMatrix, ProtoParticleVector &protoParticleVector) const
+{
+    ClusterSet usedClusters;
+    ClusterVector sortedKeyClusters;
+    overlapMatrix.GetSortedKeyClusters(sortedKeyClusters);
+
+    for (const Cluster *const pKeyCluster : sortedKeyClusters)
+    {
+        if (!pKeyCluster->IsAvailable())
+            continue;
+
+        unsigned int n0(0), n1(0);
+        MatrixType::ElementList elementList;
+        overlapMatrix.GetConnectedElements(pKeyCluster, true, elementList, n0, n1);
+
+        IteratorList iteratorList;
+        this->SelectLongElements(elementList, usedClusters, iteratorList);
+
+        // Check that elements are not directly connected and are significantly longer than any other directly connected elements
+        for (IteratorList::const_iterator iIter = iteratorList.begin(), iIterEnd = iteratorList.end(); iIter != iIterEnd; ++iIter)
+        {
+            if (TwoViewLongTracksTool::HasLongDirectConnections(iIter, iteratorList))
+                continue;
+
+            if (!TwoViewLongTracksTool::IsLongerThanDirectConnections(iIter, elementList, m_minMatchedSamplingPointRatio, usedClusters))
```

Comment:
See above/below comments about dropping passing the sampling points ratio as an argument

Reply:
Leaving this as it is